### PR TITLE
Task/split hdfs files

### LINKED ISSUE #1830: Split HDFS files by hour/day/month/year
When there are a lot of requests to a same file in HDFS the files where data is written can get really big. This makes the application slower and makes harder file processing.

In order to process the all the info without having large files, it's necessary to split files by hour/day/month/year. This can be achieved by adding a new flag into the agent.conf like the following

`cygnusagent.sinks.hdfs-sink-hdfs.file_split_periodicity= `

The possible values would be:

`
hourly
daily
monthly
yearly
`
This will add the value corresponding the flag to the file name when processing requests.

For instance.

`file_split_periodicity = hourly`

When a requests arrives at **10:59 am on 11/03/2020** and it will be added to a file **service.csv**

It will try to append all data on the request to the file **service_1011032020.csv**

if a new requests arrives at **11:00 am on 11/03/2020**

It will be appended to the file **service_1111032020.csv**

At all cases if the file doesn't exists, it must be created.

### PR DESCRIPTION
**Pull request by IvanHdzC**
Fixes #1830

### CODE REVIEW

File: CHANGES_NEXT_RELEASE [hunk 1]
```diff
@@ -3,4 +3,5 @@
 [cygnus-ngsi][PostGRESQLSink] Fix schemaName when inserting data into DB
 [cygnus-common][PostgreSQL, MySQL] SQL error persistance (#1791)
 [cygnus-ngsi][PostgisSink] Enable postgis functions without native types flag on.
-[cygnus-ngsi][RestHandler] Add a not null validation for ngsiVersion header of request.
\ No newline at end of file
+[cygnus-ngsi][RestHandler] Add a not null validation for ngsiVersion header of request.
+[cygnus-ngsi][HDFSSink] Separate files by property periodicity.
```
Comment (fgalan) [inline CHANGES_NEXT_RELEASE]:
```suggestion
[cygnus-ngsi][HDFSSink] Separate files by property periodicity (#1830)
```

Please remember to include the number of issue being fixed (in the case we have one) ;)

File: doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_hdfs_sink.md [hunk 2]
```diff
@@ -268,6 +268,7 @@ NOTE: `hive` is the Hive CLI for locally querying the data.
 | Parameter | Mandatory | Default value | Comments |
 |---|---|---|---|
 | type | yes | N/A | Must be <i>com.telefonica.iot.cygnus.sinks.NGSIHDFSSink</i> |
+| attr_metadata_store | no | true | If true, it will store metadata as usual. If false it will not store metadata regardless of the `file_format` type  |
```
Comment (fgalan) [inline doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_hdfs_sink.md]: This has been included in this PR? Or it's a gap (i.e. something developed in aprevious PR but not documented until now)?

Comment (IvanHdzC) [reply doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_hdfs_sink.md]: Just moved it a little bit up on the list.

Comment (fgalan) [reply doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_hdfs_sink.md]: Ah, ok, I didn't realize the deleted L302 line.

NTC

File: cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/sinks/NGSIHDFSSinkTest.java [hunk 3]
```diff
@@ -46,10 +46,7 @@
 
 import org.junit.Test;
 
-import java.util.ArrayList;
-import java.util.HashMap;
-import java.util.LinkedHashMap;
-import java.util.Map;
+import java.util.*;
```
Comment (manucarrace) [inline cygnus-ngsi/src/test/java/com/telefonica/iot/cygnus/sinks/NGSIHDFSSinkTest.java]: Is this a good practice?

File: CHANGES_NEXT_RELEASE (hunk 1 above)
Comment (fgalan) [reply CHANGES_NEXT_RELEASE]: Fixed in b910809

**Pull request merged**